lib/reek/smells/too_many_constants.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+require_relative 'smell_detector'
+require_relative 'smell_warning'
+
+module Reek
+  module Smells
+    #
+    # A Large Class is a class or module that has a large number of
+    # instance variables, methods, constants or lines of code.
+    #
+    # +TooManyConstants' reports classes having more than a
+    # configurable number of constants.
+    #
+    # See {file:docs/Too-Many-Constants.md} for details.
+    class TooManyConstants < SmellDetector
+      # The name of the config field that sets the maximum number
+      # of constants permitted in a class.
+      MAX_ALLOWED_CONSTS_KEY = 'max_constants'.freeze
+      DEFAULT_MAX_CONSTS = 5
+
+      def self.contexts
+        [:class, :module]
+      end
+
+      def self.default_config
+        super.merge(
+          MAX_ALLOWED_CONSTS_KEY => DEFAULT_MAX_CONSTS,
+          EXCLUDE_KEY => []
+        )
+      end
+
+      #
+      # Checks +klass+ for too many constants.
+      #
+      # @return [Array<SmellWarning>]
+      #
+      def inspect(ctx)
+        max_allowed_consts = value(MAX_ALLOWED_CONSTS_KEY, ctx)
+
+        count = ctx.each_node(:casgn).delete_if(&:constant_definition).length
```

Comment:
This makes me want to change the [`constant_definition`](https://github.com/waldyr/reek/blob/master/lib/reek/ast/sexp_extensions/module.rb#L90)